packages/billing/stripe-webhook-handlers/index.ts

```diff
@@ -130,27 +136,34 @@ export default async function stripeWebhookHandler(
 
   switch (type) {
     // These handlers should eventually become jobs which workers will process asynchronously
-    case 'invoice.payment_succeeded':
+    case 'invoice.payment_succeeded': {
       await handlePaymentSucceeded(
         dbAdapter,
         event as StripeInvoicePaymentSucceededWebhookEvent,
       );
-      sendBillingNotification(event.data.object.customer);
+      sendBillingNotification({ stripeCustomerId: event.data.object.customer });
       break;
-    case 'customer.subscription.deleted': // canceled by the user, or expired due to payment failure, or payment dispute
+    }
+    case 'customer.subscription.deleted': {
+      // canceled by the user, or expired due to payment failure, or payment dispute
       await handleSubscriptionDeleted(
         dbAdapter,
         event as StripeSubscriptionDeletedWebhookEvent,
       );
-      sendBillingNotification(event.data.object.customer);
+      sendBillingNotification({ stripeCustomerId: event.data.object.customer });
       break;
-    case 'checkout.session.completed':
+    }
+    case 'checkout.session.completed': {
       await handleCheckoutSessionCompleted(
         dbAdapter,
         event as StripeCheckoutSessionCompletedWebhookEvent,
       );
-      sendBillingNotification(event.data.object.customer);
+      await sendBillingNotification({
```

Comment:
Add code comments explaining scenarios about when stripeCustomerId or encodedMatrixUserId are present

Reply:
Perhaps it would be simpler if you introduced a function `extractUser(stripeWebhookPayload)` function and then pass the user to the function where you send the event. That way you could get rid of this parameter duality